source/_posts/routing-in-react.md

```diff
@@ -0,0 +1,387 @@
+    import {useParams} from "react-router-dom";
+    import axios from "axios";
+    
+    export const userIds = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+    
+    //Home component
+    const Home = ({props}) => {
+        const params = useParams();
+        console.log("params", params);
+        const [userId, setUserId] = React.useState();
+    
+        useEffect(() => {
+            setUserId(params.id);
+        }, [params]);
+    
+    
+        return (
+            <Grid container flexWrap={"nowrap"} spacing={2}>
+                <Grid item xs={3}>
+                    <Paper>
+                        <h2>Users</h2>
+                        {userIds.map(userId => (
+                            <Card key={userId}>
+                                <img src={`https://robohash.org/${userId}`} width={"100"} alt="user"/>
+                                <Link to={`/${userId}`}>User {userId}</Link>
+                            </Card>
+                        ))}
+                    </Paper>
+    
+                </Grid>
+                <Grid item xs={9}>
+                    {userId?<h1>User Id: {userId}</h1>:<h1>Select user</h1>
+                    }
+                </Grid>
+            </Grid>
+        );
+    };
+    
+    export default Home
+
+In this example, we have three routes defined: a home route, an about route, and a contact route. The home route uses a wildcard **path** prop (**/**) to match any path that starts with a **/**, while the about and contact routes use the **exact** prop to specify that the path must be an exact match.
+
+If a user navigates to the **/**path, the home route will be rendered, typically with a home component. If the user navigates to the **/about** path, the about route will be rendered. And if the user navigates to the **/contact** path, the contact route will be rendered.
+
+What do you do if a user tries to access a route and none of the path matches with routes defined in the Routes component?  
+We use default route!
+
+    <Route path="*" element={<Navigate to="/pathA" ... />} />
+
+#### Note
+
+React Router v6 introduces a Routes component that is kind of like the Switch component, but a lot more powerful. The main advantages of Routes over Switch are:
+
+1.  Within a <Routes>, every <Route> and <Link> is relative. This results in "Route path" and "Link to" code that is more predictable and lighter.
+    
+2.  Routes are chosen based on the best match instead of being traversed in order. This avoids bugs due to unreachable routes because they were defined later in your Switch component.
+    
+3.  Routes may be nested in one place instead of being spread out in different react components.
+    
+
+### Link and NavLink
+
+The **Link** and **NavLink** components in React Router Dom are used to create links between different routes in your application. They work similarly to the element in HTML, but they use the **to** prop to specify the target route instead of the **href** attribute.
```

Comment:
Word 'to' should not be highlighted.

Reply:
It's the to property of the component and a simple language word that's why I highlighted it. Should I not?

Reply:
that to refers to a property of the component that's why I highlighted it. Is that okay?